File: QytetetJava/src/modeloqytetet/TituloPropiedad.java

package modeloqytetet;

public class TituloPropiedad {
    private final String nombre;
    private boolean hipotecada;
    private final int alquilerBase;
    private final float factorRevalorizacion;
    private final int hipotecaBase;
    private final int precioEdificar;
    private Casilla casilla;
    private Jugador propietario;
    
    public TituloPropiedad(String nom, int alquiler, float factor, int hip, int precioE) {
        nombre = nom;
        hipotecada = false;
        alquilerBase = alquiler;
        factorRevalorizacion = factor;
        hipotecaBase = hip;
        precioEdificar = precioE;
    }
    
    void cobrarAlquiler(int coste) {
        propietario.modificarSaldo(-coste);
    }

    public String getNombre() {
        return nombre;
    }

    boolean getHipotecada() {
        return hipotecada;
    }

    int getAlquilerBase() {
        return alquilerBase;
    }

    float getFactorRevalorizacion() {
        return factorRevalorizacion;
    }

    int getHipotecaBase() {
        return hipotecaBase;
    }

    int getPrecioEdificar() {
        return precioEdificar;
    }
    
    Jugador getPropietario() {
        return propietario;
    }
    
    public Casilla getCasilla() {
        return casilla;
    }
    
    boolean propietarioEncarcelado() {
        return propietario.getEncarcelado();
    }

    void setHipotecada(boolean hipotecada) {
        this.hipotecada = hipotecada;
    }
    
    void setCasilla(Casilla casilla) {
        this.casilla = casilla;
    }
    
    void setPropietario(Jugador propietario) {
        this.propietario = propietario;
    }
    
    public boolean tengoPropietario() {
        return propietario != null;
    }

    @Override
    public String toString() {
        return "TituloPropiedad{" + "nombre=" + nombre + ", hipotecada=" + hipotecada + ", alquilerBase=" + alquilerBase + ", factorRevalorizacion=" + factorRevalorizacion + ", hipotecaBase=" + hipotecaBase + ", precioEdificar=" + precioEdificar + '}';
    }
    
    
}

File: QytetetRuby/lib/titulo_propiedad.rb
#encoding: utf-8
require_relative "jugador"
require_relative "casilla"

module ModeloQytetet
  class TituloPropiedad
    attr_accessor :hipotecada, :casilla, :propietario
    attr_reader :nombre, :alquilerBase, :factorRevalorizacion, :hipotecaBase, :precioEdificar
    
    def initialize(nom, alquiler, factor, hip, precioE)
      @nombre = nom
      @hipotecada = false
      @alquilerBase = alquiler
      @factorRevalorizacion = factor
      @hipotecaBase = hip
      @precioEdificar = precioE
    end
    
    def cobrar_alquiler(coste)
      @propietario.modificar_saldo(-coste)
    end
    
    def get_nombre
      @nombre
    end
    
    def get_hipotecada
      @hipotecada
    end
    
    def get_alquiler_base
      @alquilerBase
    end
    
    def get_factor_revalorizacion
      @factorRevalorizacion
    end
    
    def get_hipoteca_base
      @hipotecaBase
    end
    
    def get_precio_edificar
      @precioEdificar
    end
    
    def get_propietario
      @propietario
    end
    
    def propietario_encarcelado
      @propietario.get_encarcelado
    end
    
    def set_hipotecada(hipotecada)
      @hipotecada = hipotecada
    end
    
    def set_casilla(casilla)
      @casilla = casilla
    end
    
    def set_propietario(propietario)
      @propietario = propietario
    end
    
    def tengo_propietario
      @propietario != nil
    end
        
    def to_s
      "TituloPropiedad { nombre = #{@nombre}, hipotecada = #{@hipotecada}, alquilerBase = #{@alquilerBase}, factorRevalorizacion = #{@factorRevalorizacion}, hipotecaBase = #{@hipotecaBase}, precioEdificar = #{@precioEdificar}"
    end
  end
end
